src/booster.rs

```diff
@@ -274,29 +274,72 @@ impl Booster {
     /// Get names of evaluation metrics
     pub fn eval_names(&self) -> Result<Vec<String>> {
         let num_metrics = self.num_eval()?;
-        let metric_name_length = 32;
+
+        /////////////////////////////////////////////////////////////////////
+        // call with 0-sized buffer to find out how much space to allocate
+        /////////////////////////////////////////////////////////////////////
         let mut num_eval_names = 0;
         let mut out_buffer_len = 0;
-        let out_strs = (0..num_metrics)
-            .map(|_| {
-                CString::new(" ".repeat(metric_name_length))
-                    .unwrap()
-                    .into_raw() as *mut c_char
-            })
+
+        lgbm_call!(lightgbm_sys::LGBM_BoosterGetEvalNames(
+            self.handle,
+            0,
+            &mut num_eval_names,
+            0,
+            &mut out_buffer_len,
+            std::ptr::null_mut() as *mut *mut c_char
+        ))
+        .unwrap();
+
+        /////////////////////////////////////////////////////////////////////
+        // sanity check
+        /////////////////////////////////////////////////////////////////////
+        assert_eq!(num_eval_names, num_metrics);
```

Comment:
Maybe return an Error instead of panicking at this point?

Reply:
Hm, good point. A failure here is in "this should never go wrong" territory, but not in "the rest of my program might be broken too" territory.